src/tools/illink/src/ILLink.Shared/TrimAnalysis/ValueExtensions.cs

```diff
@@ -50,10 +52,20 @@ internal static string ValueToString (this SingleValue value, params object[] ar
 
 		internal static SingleValue? AsSingleValue (this in MultiValue node)
 		{
-			if (node.Count () != 1)
+			var values = node.AsEnumerable ();
+			if (values.Count () != 1)
 				return null;
 
-			return node.Single ();
+			return values.Single ();
+		}
+
+		// ValueSet<TValue> is not enumerable. This helper translates ValueSet<SingleValue>.Unknown
+		// into a ValueSet<SingleValue> whose sole element is UnknownValue.Instance.
+		internal static IEnumerable<SingleValue> AsEnumerable (this MultiValue multiValue)
+		{
+			return multiValue.IsUnknown ()
+				? ImmutableArray.Create (UnknownValue.Instance)
```

Comment:
This will allocate on every call - would be better to have this precached as a static field.